File: Python/IsSubsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        l=0
        r=0
        while l<len(s) and r<len(t):
            if s[l]==t[r]:
                l+=1
            r+=1
        return l==len(s)
    
obj=Solution()

t=input("t : ")
s=input("s : ")
res=obj.isSubsequence(s,t)
print(f"Is subsequence : {res}")

# Input: s = "abc", t = "ahbgdc"
# Output: true

# Input: s = "axc", t = "ahbgdc"
# Output: false

File: Java/IsSubsequence.java
import java.util.*;

public class IsSubsequence {
    public boolean isSubsequence(String s, String t) {
        int r=0;
        int l=0;
        while (l<s.length() && r<t.length()){
            if(s.charAt(l) == t.charAt(r)){
                l++;
            }
            r++;
        }
        return l==s.length();
    }
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        IsSubsequence obj=new IsSubsequence();
        System.out.print("s : ");
        String s=sc.nextLine();
        System.out.print("t : ");
        String t=sc.nextLine();
        System.out.println("Is Subsequence : " + obj.isSubsequence(s, t));

    }
}